src/OrchardCore.Modules/OrchardCore.Menu/MenuShapes.cs

```diff
@@ -102,8 +135,39 @@ public override ValueTask DiscoverAsync(ShapeTableBuilder builder)
 
                 if (menuItems != null)
                 {
+                    IEnumerable<Permission> allPermissions = null;
```

Comment:
Move to private method (HasPermissions ...)

Reply:
done